lib/std/posix.zig

```diff
@@ -192,10 +192,18 @@ pub const iovec_const = extern struct {
     len: usize,
 };
 
-pub const ACCMODE = enum(u2) {
-    RDONLY = 0,
-    WRONLY = 1,
-    RDWR = 2,
+pub const ACCMODE = switch (native_os) {
+    // https://github.com/SerenityOS/serenity/blob/4adc51fdf6af7d50679c48b39362e062f5a3b2cb/Kernel/API/POSIX/fcntl.h#L28-L30
+    .serenity => enum(u2) {
+        RDONLY = 1,
+        WRONLY = 2,
+        RDWR = 3,
+    },
```

Comment:
I tried to change this upstream to match the rest of the universe but it was non-trivial and ultimately decided against: https://github.com/SerenityOS/serenity/pull/25804

Reply:
There's an interesting note in POSIX about this: https://pubs.opengroup.org/onlinepubs/007904875/functions/open.html

> In historical implementations the value of O_RDONLY is zero. Because of that, it is not possible to detect the presence of O_RDONLY and another option. Future implementations should encode O_RDONLY and O_WRONLY as bit flags so that:
>
> O_RDONLY | O_WRONLY == O_RDWR

...which matches what serenity does so at least there's that :slightly_smiling_face: